package org.hiero.base.test.config;

import java.util.Optional;
import java.util.Set;
import org.hiero.base.config.ConsensusNode;
import org.hiero.base.config.NetworkSettings;
import org.jspecify.annotations.NonNull;

public class SoloActionNetworkSettings implements NetworkSettings {

  @Override
  public @NonNull String getNetworkIdentifier() {
    return "hiero-solo-action";
  }

  @Override
  public @NonNull Optional<String> getNetworkName() {
    return Optional.of("Hiero Solo Action");
  }

  @Override
  public @NonNull Set<String> getMirrorNodeGrpcAddresses() {
    return Set.of("localhost:5600");
  }

  @Override
  public @NonNull Optional<String> getMirrorNodeRestUrl() {
    return Optional.of("http://localhost:38081");
  }

  @Override
  public @NonNull Set<ConsensusNode> getConsensusNodes() {
    return Set.of(new ConsensusNode("127.0.0.1", "35211", "0.0.3"));
  }

  @Override
  public @NonNull Optional<Long> chainId() {
    return Optional.empty();
  }

  @Override
  public @NonNull Optional<String> relayUrl() {
    return Optional.of("http://localhost:37546"); // JSON-RPC-Relay port from hiero-solo-action
  }
}
